Application/Source/Detail/View/DetailViewModel.swift

```diff
@@ -50,24 +48,26 @@ class DetailViewModel: ViewModel, SelectionPresentingViewModel {
                 .disposed(by: self.disposeBag)
         })
 
-    init(selectionFactory: SelectionViewModelFactoryProtocol) {
+    init(selectionFactory: SelectionViewModelFactoryProtocol, foodTableFactory: FoodTableViewModelFactoryProtocol) {
```

Comment:
Do you think this should be a single `factory: SelectionViewModelFactoryProtocol & FoodTableViewModelFactoryProtocol` (perhaps typealias'd as `Factory` in the scope of this `ViewModel`) parameter instead? I can't imagine a case where you would be passing two separate instances here...